compiler/qsc_parse/src/item.rs

```diff
@@ -136,15 +136,15 @@ fn parse_namespace(s: &mut ParserContext) -> Result<Namespace> {
     let lo = s.peek().span.lo;
     let doc = parse_doc(s).unwrap_or_default();
     token(s, TokenKind::Keyword(Keyword::Namespace))?;
-    let name = dot_ident(s)?;
+    let name = path(s)?;
     token(s, TokenKind::Open(Delim::Brace))?;
     let items = barrier(s, &[TokenKind::Close(Delim::Brace)], parse_many)?;
     recovering_token(s, TokenKind::Close(Delim::Brace));
     Ok(Namespace {
         id: NodeId::default(),
         span: s.span(lo),
         doc: doc.into(),
-        name,
+        name: (*name).into(),
```

Comment:
I admit I'm not a huge fan of having a `From<T>`  implementation for each type we may convert the `Idents` to, defined alongside `Idents`.

This is one example. I see this is coming from `impl From<Path>`  defined in `ast.rs`. It's three lines of code that's only used here, that could easily just live in this function. It makes the code less readable to see `into()` here, have to check the types, and then go search in a different file to understand what's actually going on (which is a nontrivial conversion).

Reply:
Discussed offline. We agreed that we may not even need to use the `Path` parser for `Idents`, which will eliminate the need for a `From<Path>` impl, making this conversation moot. We'll take this in a future PR since it'll involve the parser changes from #1469 for parsing dotted identifiers.